templates/demo-store/.env

```diff
@@ -3,4 +3,4 @@
 SESSION_SECRET="foobar"
 PUBLIC_STOREFRONT_API_TOKEN="3b580e70970c4528da70c98e097c2fa0"
 PUBLIC_STOREFRONT_API_VERSION="2023-01"
-PUBLIC_STORE_DOMAIN="hydrogen-preview.myshopify.com"
+PUBLIC_STORE_DOMAIN="checkout.hydrogen.shop"
```

Comment:
Domain routing is done with Shopify admin

<img width="874" alt="Screenshot 2023-03-02 at 4 06 44 PM" src="https://user-images.githubusercontent.com/2319002/222596579-36c20f41-2375-4b99-b5db-0d461c132040.png">

This is to establish the sub domain environment. When the primary Online Store domain is point to a sub domain (`checkout.hydrogen.shop`) of the `hydrogen.shop`, the following things will happen:

* SFAPI graphql endpoint will also work for the 3P domain assigned (`checkout.hydrogen.shop/api/2023-01/gragql.json`)
* Shopify cookies will be propagated properly because we have the same top-level domain (`hydrogen.shop`)
* Checkout url generated from SFAPI will also be point to 3P domain (`checkout.hydrogen.shop`) thus removes the auto-redirect from `hydrogen-preview.myshopify.com` to `checkout.hydrogen.shop`

Reply:
`hydrogen-preview.myshopify.com` still works. In fact, you can still use it to make api calls as well. However, it will have performance impacts:

* Longer api times - due to domain redirect
* Breaking of Shopify cookies - due to domain redirect